### PR TITLE
Feature/create classes

### PR DESCRIPTION
**Pull request by vechicin**
# In this feature I did the following:

1. Created a class Person with the following:

- Instance vars: @id, @name, and @age.

- Constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have default values of "Unknown" and true.

- Getters for @id, @name, and @age.

- Setters for @name and @age.

- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.

- Public method can_use_services? that returns true if person is of age or if they have permission from parents.

2. Created a class Student with the following:

- Inherits from Person.

- Constructor extends parent's constructor by adding @classroom and a parameter for it.

- Method play_hooky that returns "¯\(ツ)/¯".

3. Created a class Teacher with the following:

- Inherits from Person.

- Constructor extends parent's constructor by adding @specialization and a parameter for it.

- Override can_use_services? so it always returns true.

4. Each class is saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    return false unless is_of_age? || @parent_permission
+
+    true
```
Comment (mmsesay) [inline person.rb]: - [ ] Good approach but I think it can be more simplified by a one-liner `is_of_age?() || @parent_permission`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
```
Comment (mmsesay) [inline person.rb:21]: - [ ] Good check :+1: . Ruby is clean in terms of syntax. Simplicity makes it a gem  `@age >= 18` :gem:

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @vechicin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi @vechicin, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**